File: app/src/main/java/africa/of/designers/churchill/LessonDetailActivity.java
package africa.of.designers.churchill;

import android.content.Intent;
import android.support.design.widget.FloatingActionButton;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.view.Window;
import android.view.WindowManager;
import android.widget.AdapterView;
import android.widget.ArrayAdapter;
import android.widget.ImageView;
import android.widget.ListView;
import android.widget.TextView;

import com.parse.ParseUser;

import java.util.ArrayList;
import java.util.List;


public class LessonDetailActivity extends AppCompatActivity {

    private ImageView MovieThumbnailImg,MovieCoverImg;
    private TextView tv_title,tv_description;
    private FloatingActionButton play_fab;
    private String movieTitle;
    private ListView ActorsRV ;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        // remove title
        requestWindowFeature(Window.FEATURE_NO_TITLE);
        getWindow().setFlags(WindowManager.LayoutParams.FLAG_FULLSCREEN,
                WindowManager.LayoutParams.FLAG_FULLSCREEN);

        setContentView(R.layout.activity_movie_detail);
        // ini views
        iniViews();

        ActorsRV = (ListView) findViewById(R.id.Rv_actors);


        List<String> list = new ArrayList<>();

        if (movieTitle.equals("Alphabets")){
            list.add("$5 Coming soon");
        }else{
            list.add("Start Taking Lessons");
        }

        ActorsRV.setAdapter(new ArrayAdapter<String>(this, android.R.layout.simple_list_item_1, list));
        ActorsRV.setOnItemClickListener(new AdapterView.OnItemClickListener() {
            @Override
            public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
                if (position == 0){
                    if(movieTitle.equals("Greetings/Conversations")) {
                        if (ParseUser.getCurrentUser() != null) {

                                startActivity(new Intent(LessonDetailActivity.this, GreetingsActivity.class));
                        }
                    }
                    if(movieTitle.equals("Family")) {
                        if (ParseUser.getCurrentUser() != null) {

                                startActivity(new Intent(LessonDetailActivity.this, FamilyActivity.class));
                        }
                    }
                    if(movieTitle.equals("Counting")) {
                        if (ParseUser.getCurrentUser() != null) {

                                startActivity(new Intent(LessonDetailActivity.this, CountingActivity.class));
                        }
                    }
                    if(movieTitle.equals("Days/Festive Periods/Seasons")) {
                        if (ParseUser.getCurrentUser() != null) {

                                startActivity(new Intent(LessonDetailActivity.this, SeasonsActivity.class));
                        }
                    }
                    if(movieTitle.equals("Calendar")) {
                        startActivity(new Intent(LessonDetailActivity.this, CalendarActivity.class));
                    }
                }
            }
        });

    }

    void iniViews() {

        movieTitle = getIntent().getExtras().getString("title");
        int imageResourceId = getIntent().getExtras().getInt("imgURL");
        int imagecover = getIntent().getExtras().getInt("imgCover");
        int desc = getIntent().getExtras().getInt("imgDesc");
        MovieThumbnailImg = findViewById(R.id.detail_movie_img);

        MovieThumbnailImg.setImageResource(imageResourceId);
        MovieCoverImg = findViewById(R.id.detail_movie_cover);

        tv_title = findViewById(R.id.detail_movie_title);
        MovieCoverImg.setImageResource(imagecover);
        try {
            tv_title.setText(movieTitle);
        }catch (Exception e19){
            e19.printStackTrace();
        }

        tv_description = findViewById(R.id.detail_movie_desc);
        tv_description.setText(desc);

    }


}

File: app/src/main/java/africa/of/designers/churchill/ui/lessondetail/LessonDetailActivity.kt
package africa.of.designers.churchill.ui.lessondetail

import africa.of.designers.churchill.*
import africa.of.designers.churchill.base.BaseActivity
import africa.of.designers.churchill.data.model.ExtrasInt
import africa.of.designers.churchill.databinding.ActivityLessonDetailBinding
import africa.of.designers.churchill.ui.lesson.LessonActivity
import africa.of.designers.churchill.util.startNewActivityWithExtra
import android.os.Bundle
import android.view.Window
import android.view.WindowManager
import android.widget.*
import android.widget.AdapterView.OnItemClickListener
import androidx.databinding.DataBindingUtil
import androidx.lifecycle.ViewModelProvider
import kotlinx.android.synthetic.main.activity_lesson_detail.*
import kotlinx.android.synthetic.main.activity_main.*
import java.util.ArrayList

class LessonDetailActivity : BaseActivity(), LessonDetailListener {
    private var imageRes: Int? = null
    private var imageCoverRes: Int? = null
    private var desc: Int? = null
    private var lessonId: Int? = null
    private var lessonTitle: String? = null

    lateinit var binding : ActivityLessonDetailBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        // remove title
        requestWindowFeature(Window.FEATURE_NO_TITLE)
        window.setFlags(
            WindowManager.LayoutParams.FLAG_FULLSCREEN,
            WindowManager.LayoutParams.FLAG_FULLSCREEN
        )

        binding = DataBindingUtil.setContentView(this, R.layout.activity_lesson_detail)
        val viewModel = ViewModelProvider(this)[LessonDetailViewModel::class.java]
        binding.lessonviewmodel = viewModel
        viewModel.lessonListener = this

        iniViews()
        initializeList()
    }

    private fun iniViews() {
        lessonTitle = intent.extras!!.getString("title")
        lessonId = intent.extras!!.getInt("lessonId")
        imageRes = intent.extras!!.getInt("imgURL")
        imageCoverRes = intent.extras!!.getInt("imgCover")
        desc = intent.extras!!.getInt("imgDesc")

        binding.lessonImg.setImageResource(imageRes!!)
        binding.lessonCoverImg.setImageResource(imageCoverRes!!)
        binding.lessonTitle.text = lessonTitle
        binding.lessonDesc.setText(desc!!)
    }

    private fun initializeList() {
        val list: MutableList<String> = ArrayList()
        list.add("Start Taking Lessons")

        type_list.apply {
            adapter = ArrayAdapter(this@LessonDetailActivity, android.R.layout.simple_list_item_1, list)
            onItemClickListener = OnItemClickListener { _, _, position, _ ->
                if (position == 0) {
                    val extraData: ArrayList<ExtrasInt> = ArrayList()
                    extraData.add(ExtrasInt("id", lessonId!!))
                    startNewActivityWithExtra(LessonActivity::class.java, false, null, extraData, null)
                }
            }
        }
    }
}
